src/components/order-service/order-service.controller.ts

```diff
@@ -72,7 +93,34 @@ export class OrderServiceController implements angular.IController {
     return;
   }
 
-  public cancelOrder() {
-    return;
+  public closePanel() {
+    if (angular.isFunction(this.ctrl.handleClose)) {
+      this.ctrl.handleClose();
+    }
+  }
+
+  private listProjects() {
+    this.DataService.list('projects', this.$scope).then((response: any) => {
+      this.ctrl.projects = _.sortBy(response.by('metadata.name'), this.$filter('displayName'));
+      this.ctrl.selectedProject = _.first(this.ctrl.projects);
+    });
+  }
+
+  private makeServiceInstance() {
+    let serviceClassName = _.get(this, 'ctrl.serviceClass.resource.metadata.name');
+    let serviceInstance = {
+      kind: 'Instance',
+      apiVersion: 'servicecatalog.k8s.io/v1alpha1',
+      metadata: {
+        namespace: this.ctrl.selectedProject.metadata.name,
+        generateName: serviceClassName + '-'
```

Comment:
From my reading of the godoc, the prefix will be truncated if necessary.

https://github.com/spadgett/origin/blob/9b26d5a134bd12767ab7a466c44aeef19a1af13a/vendor/k8s.io/client-go/pkg/api/types.go#L68-L73

> The provided value must be valid within the rules for Name, and may be truncated by the length of the suffix required to make the value unique on the server.